src/aspire/abinitio/commonline_sync3n.py

```diff
@@ -820,6 +823,143 @@ def _estimate_all_Rijs(self, clmatrix):
         """
         Estimate Rijs using the voting method.
 
+        :param clmatrix: Common lines matrix
+        :return: Estimated rotations
+        """
+        # host/gpu dispatch
+        if self.__gpu_module:
+            res = self._estimate_all_Rijs_cu(clmatrix)
+        else:
+            res = self._estimate_all_Rijs_host(clmatrix)
+
+        return res
+
+    def _estimate_all_Rijs_cu(self, clmatrix):
+        import cupy as cp
+
+        estimate_all_angles1 = self.__gpu_module.get_function("estimate_all_angles1")
+        estimate_all_angles2 = self.__gpu_module.get_function("estimate_all_angles2")
+        angles_to_rots = self.__gpu_module.get_function("angles_to_rots")
+
+        # Use the sync3n MATLAB implementation,
+        # other mode exists to support other CL methods.
+        sync = 1
+
+        # transfer input to device
+        clmatrix = cp.asarray(clmatrix, order="C", dtype=np.int16)
+
+        # workspace arrays
+        ntics = int(180 / self.hist_bin_width)
+        n_pairs = self.n_img * (self.n_img - 1) // 2
+        hist = cp.zeros((self.n_img, ntics), dtype=np.float64)
+        # k_map stores the mapping of i, k indices to histogram bins
+        k_map = cp.zeros((self.n_img, self.n_img), dtype=np.uint16)
+        # angles_map stores mapping of i, k indices to angles
+        angles_map = cp.zeros((self.n_img, self.n_img), dtype=np.float64)
+        # resulting pairs i,j euler angles
+        angles = cp.zeros((n_pairs, 3), dtype=np.float64)
+
+        # Configure 2d grid of blocks (kernel 1)
+        blkszx = 32
+        nblkx = (self.n_img + blkszx - 1) // blkszx  # i
+        blkszy = 32
+        nblky = (self.n_img + blkszy - 1) // blkszy  # k
+
+        # Configure 1d grid of blocks (kernel 2)
+        blksz = 1024
+        nblk = (self.n_img + blksz - 1) // blksz
+
+        from time import perf_counter
```

Comment:
Do we want to leave this in production? Fine with me, but I haven't seen it elsewhere in our code.

Reply:
I was just code-blind to these lines;  It was more for my development. 

Logs should be good enough for now. Removed, thanks.